### ISSUE TITLE
REQUEST FOR HELP - TEXTURE / MODEL TWEAKS

### ISSUE THREAD

**Issue by radfast** [micdoodle8/Galacticraft#3272]
Some of you guys are _much_ better than me with 3D models and textures...

Galacticraft could really do with your help fixing a few minor issues:

*  [Tier 1 Rocket](https://github.com/micdoodle8/Galacticraft/blob/master/src/main/resources/assets/galacticraftcore/models/rocket_t1.obj) - z-fighting at the top

*  [Tier 2 Rocket](https://github.com/micdoodle8/Galacticraft/blob/master/src/main/resources/assets/galacticraftplanets/models/rocket_t2.obj) - some gaps where rendered faces don't join properly

*  [NASA workbench](https://github.com/micdoodle8/Galacticraft/blob/master/src/main/resources/assets/galacticraftcore/models/block/workbench.obj) - z-fighting on one of the arms

*  [Moon Buggy](https://github.com/micdoodle8/Galacticraft/blob/master/src/main/resources/assets/galacticraftcore/models/buggy.obj) - the steps on the right hand side of the buggy have a "twisted" texture (incorrect uv coordinates) and the buggy model has a few other small uv errors

You can find the Wavefront `.obj` files linked above, and there are `.mtl` material files in the same folders.  The textures are in `assets/textures`, filenames and relative paths given in the material files.

I guess what I'm hoping for is a corrected `.obj` file - something like the current file but with small adjustments to texture uvs or vertex coordinates.

--------

For the NASA workbench, I've also been sent these two images which are supposed to show a texture inversion error, but I'm actually not able to see it in-game, maybe you guys can see it?

![dy5qcxpx4aabuax](https://user-images.githubusercontent.com/6842258/37805253-e8abee2a-2e30-11e8-8acc-42b66ed2b58c.jpg)
![dy5qegkx0aem4tn](https://user-images.githubusercontent.com/6842258/37805259-ecb0d3c8-2e30-11e8-999b-c37015ee5a22.jpg)

**Comment by EzerArch** [comment]
<*i'm doing my part.gif*>

https://twitter.com/EzerArch/status/977749774500794368

**Comment by marcus8448** [comment]
Well, I tried... 😐 
![2018-03-27_20 29 46](https://user-images.githubusercontent.com/28928887/38007337-d1c7d266-31fd-11e8-8db5-fe96b24907e5.png)
I'm not so good at 3d models...

**Comment by EzerArch** [comment]
What you're saying? The invisible cloak for rockets is excellent!

**Comment by kouwei32** [comment]
I'm going to try to fix some of this, but I have no experience with 3-D models. Let's see how this turns out. 🙃

**Comment by EzerArch** [comment]
Try Blender and Metasequoia.

**Comment by Dominance** [comment]
Yeah on the screen shots you tossed up it is pretty easy to see that the model is not scaled to pixel. I just kinda stumbled upon this issue, you all still needing help?

**Comment by radfast** [comment]
> you all still needing help?

@Dominance  Yes please!  Help fixing even one of these issues would be awesome.  (Will give me more time to work on [this](https://github.com/micdoodle8/Galacticraft/commit/d31473df61c0d59c045e1611106a7f293ae1f9ed#commitcomment-28789626)).

**Comment by Dominance** [comment]
Ok I will see what I can do and toss up a PR

**Comment by Dominance** [comment]
So is there a reason why the textures used are so large? https://github.com/micdoodle8/Galacticraft/blob/master/src/main/resources/assets/galacticraftcore/textures/model/buggy.png For example is 512x256 and it would seem as if all of these could easily fit in a much smaller texture. I can redo the UV locations and make them fit into a smaller texture if it would help performance.

**Comment by radfast** [comment]
There's no reason, only that that's how the person / software who originally made that model did it.

BUT:  Textures must - where possible - be sizes like 64x64, 128x128 or 256x256.   i.e. squares and powers of 2.

Two reasons for this:

1.  Some (older) graphics cards require this, and will show white for any texture not meeting this rule (actually the 512x256 texture already fails this rule)

2.  Minecraft combines all the textures in the game into a "texture atlas".  Like a game of Tetris, it will be able to fit them together better (allowing a smaller texture atlas) if they are regular sizes, so 512x256 meets this rule, but 256x256 would be even better.

TL;DR:   If you can fit it into 256x256 that would be excellent.  If you can't fit it into that size, then 512x256 is the "next best".  But there's no performance saving making it, for example, 400x256.

Bear in mind also that changing the UV locations is going to be very time-consuming for you, for no change which players will notice.  Can I suggest better to spend your time tracking down the UV errors and other graphical glitches which players do see.

BTW if you are able to edit files while the game is running (i.e. in Eclipse or another development environment) you should be able to see the effects of your work in-game if you hit `F3-T` to reload all resources.  Otherwise I guess you'll have to edit, drop the edited files into your copy of Galacticraft .jar, then launch Minecraft.

**Comment by Dominance** [comment]
Good thing is, I have pretty extensive knowledge when it comes to making obj models for mine craft. Stuff like this http://i.imgur.com/2314pvs.gifv Bad news is it would prob be easier to just remake the whole model the right way rather than to attempt to fix it. Stuff like this https://i.imgur.com/Y5lnXjz.png shows that the model/UV's where never properly scaled to pixel size in the first place. Its not a easy task to do and would take just as much effort to fix a model rather than to just remake it. To fix z fighting such as https://i.imgur.com/OVF56NZ.png you have to adjust the vertices positions which effects the UV's ether way. Do you have a discord channel i can hop into?

**Comment by radfast** [comment]
https://discordapp.com/channels/261150906429472768/261422483100532736
